testing/trino-testing/src/main/java/io/trino/testing/datatype/CreateAndInsertDataSetup.java

```diff
@@ -81,6 +83,13 @@ protected String tableDefinition(List<ColumnSetup> inputs)
                     .collect(joining(",\n", "(\n", ")"));
         }
 
+        // This should use smarter check of underlying system as soon as tests need to use e.g. `getQueryRunner()::execute` as an SqlExecutor.
+        checkState(
+                sqlExecutor instanceof TrinoSqlExecutor,
+                "Explicit declared types are required when creating a table in a system other than Trino. " +
+                        "This is to avoid ambiguities that could occur if same-looking literal has a different type " +
+                        "in various systems we connected to in tests.");
```

Comment:
I'd say this instead:
> Explicit declared input types are required when creating a table from any non-Trino system to prevent confusion between different systems' type literals.

Reply:
nice!